### PR TITLE
Updates alert rules for new teams

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Towards https://github.com/giantswarm/giantswarm/issues/19143

This PR:

- updates alert rules for new teams

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Check for duplicated alert names
    - [x] DeploymentNotSatisfiedPhoenix
- [x] Check phoenix routing and rotation
- [x] Check honeybadger routing and rotation
- [x] Check cabbage routing and rotation
- [x] Check rocket routing and rotation
- [ ] ~~Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).~~
- [ ] ~~Alerting rules must have a comment documenting why it needs to exist.~~

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/app.rules.yml [hunk 1]
```diff
@@ -27,11 +27,11 @@ spec:
         sig: none
         team: atlas
         topic: releng
-    - alert: ManagementClusterAppFailedBatman
+    - alert: ManagementClusterAppFailedHoneybadger
       annotations:
         description: '{{`Management Cluster App {{ $labels.name }}, version {{ $labels.version }} is {{if $labels.status }} in {{ $labels.status }} state. {{else}} not installed. {{end}}`}}'
         opsrecipe: app-failed/
-      expr: app_operator_app_info{status!~"(?i:(deployed|cordoned))", catalog=~"control-plane-.*",team=~"batman|halo"}
+      expr: app_operator_app_info{status!~"(?i:(deployed|cordoned))", catalog=~"control-plane-.*",team=~"batman|biscuit|halo|honeybadger"}
```
Comment (MarcelMue) [inline helm/prometheus-rules/templates/alerting-rules/app.rules.yml:34]: Why keep the old teams here? Will this not cause a lot of overlap?

File: helm/prometheus-rules/templates/alerting-rules/deployment.management-cluster.rules.yml [hunk 2]
```diff
@@ -184,7 +170,7 @@ spec:
       annotations:
         description: '{{`Deployment {{ $labels.namespace}}/{{ $labels.deployment }} is not satisfied.`}}'
         opsrecipe: deployment-not-satisfied/
-      expr: kube_deployment_status_replicas_unavailable{cluster_type="management_cluster", deployment=~"worker-.+|master-.+"} > 0
+      expr: kube_deployment_status_replicas_unavailable{cluster_type="management_cluster", deployment=~"worker-.+|master-.+|kyverno.*"} > 0
```
Comment (MarcelMue) [inline helm/prometheus-rules/templates/alerting-rules/deployment.management-cluster.rules.yml:173]: Why was Kyverno added here?

File: helm/prometheus-rules/templates/alerting-rules/draughtsman.rules.yml [hunk 3]
```diff
@@ -20,7 +20,7 @@ spec:
       labels:
         area: empowerment
         severity: notify
-        team: biscuit
+        team: honeybadger
```
Comment (MarcelMue) [inline helm/prometheus-rules/templates/alerting-rules/draughtsman.rules.yml:23]: IMO this alert should just be removed?

**Review by MarcelMue — COMMENTED** [review]
Generally okay - not sure if this was the easiest way to do it :D

File: helm/prometheus-rules/templates/alerting-rules/deployment.management-cluster.rules.yml (hunk 2 above)
Comment (JosephSalisbury) [reply helm/prometheus-rules/templates/alerting-rules/deployment.management-cluster.rules.yml:173]: it was the only deployment in `DeploymentNotSatisfiedLudacris` which I killed - needed to go into either phoenix or rocket, so i chose rocket cause you're closer to that team (we can move it to zachurity in future)

File: helm/prometheus-rules/templates/alerting-rules/draughtsman.rules.yml (hunk 3 above)
Comment (JosephSalisbury) [reply helm/prometheus-rules/templates/alerting-rules/draughtsman.rules.yml:23]: up to teams to kill any dumb alerts, this PR is already too big :D

File: helm/prometheus-rules/templates/alerting-rules/deployment.management-cluster.rules.yml (hunk 2 above)
Comment (MarcelMue) [reply helm/prometheus-rules/templates/alerting-rules/deployment.management-cluster.rules.yml:173]: Okay - it should go with me to `honeybadger` for now. We use it for releases in CAPI

**Comment by anvddriesch** [comment]
Can we merge this?

**Comment by JosephSalisbury** [comment]
@anvddriesch see https://github.com/giantswarm/giantswarm/issues/19447, @ubergesundheit is working on this today, hoping to merge this tomorrow or so

**Comment by anvddriesch** [comment]
Thanks for the heads up. :)
We had an issue where `nginx-ingress-controller` and `calico` were both failing on a MC and we did not get alerted. We suspect that this was because `batman` and `ludacris` are still assigned to these apps. Or does someone still receive pages for those teams currently?

**Comment by marians** [comment]
It seems as if alerts for e. g. calico and nginx-ingress-controller don't reach any oncaller currently because they still go to ludacris and batman. As a result, a customer had to inform us about their management cluster being unreachable. 

Looks urgent to me to get this sorted soon.

**Comment by JosephSalisbury** [comment]
As far as I'm aware, batman and ludacris are still running schedules. Let's move to #oncall in Slack for debugging anything else further.

**Review by TheoBrigitte — APPROVED** [review]

**Pull request merged**